curl-helper.c

```diff
@@ -641,14 +643,16 @@ static void resetOcamlValues(Connection* connection)
     int i;
 
     for (i = 0; i < OcamlValuesSize; i++)
-        Store_field(connection->ocamlValues, i, Val_unit);
+      if (i != Ocaml_CALLBACKBUFFER)
+            Store_field(connection->ocamlValues, i, Val_unit);
 }
 
 static Connection* allocConnection(CURL* h)
 {
     Connection* connection = (Connection *)malloc(sizeof(Connection));
 
     connection->ocamlValues = caml_alloc(OcamlValuesSize, 0);
+    Store_field(connection->ocamlValues, Ocaml_CALLBACKBUFFER, caml_ba_alloc_dims(CAML_BA_UINT8 | CAML_BA_C_LAYOUT | CAML_BA_EXTERNAL, 1, NULL, 0));
```

Comment:
it means bigarray is always allocated even if not used. Probably not a big deal but slightly meh? IIRC GC rate is computed from the number of custom values, so it might have some unexpected impact?

Reply:
You're right, better if it's only allocated when needed. Done.